joblib/externals/loky/backend/__init__.py

```diff
@@ -9,8 +9,13 @@ def _make_name():
         name = '/loky-%i-%s' % (os.getpid(), next(synchronize.SemLock._rand))
         return name
 
-    # monkey patch the name creation for multiprocessing
-    from multiprocessing import synchronize
-    synchronize.SemLock._make_name = staticmethod(_make_name)
+    # Handle cases where semaphores are not supported
+    try:
+        # monkey patch the name creation for multiprocessing
+        from multiprocessing import synchronize
+        synchronize.SemLock._make_name = staticmethod(_make_name)
+    except ImportError as import_error:
+        # sys.stderr.write(f"Disabling loky support: {import_error}")
```

Comment:
I'm not certain how this should be handled. Where is it best to log this error?

Reply:
Why is this needed if joblib/parallel.py already uses a try / except ImportError on `from .externals import loky`?